File: DsaWithCpp/Recursion/Factorial.cpp
#include <iostream>
using namespace std;

int factorial(int n);

int main()
{
    int n;
    cout << "Enter the Number -> ";
    cin >> n;
    printf("Factorial of '%d' is -> '%d'\n", n, factorial(n));
    return 0;
}

int factorial(int n)
{
    if (n == 0 || n == 1)
        return 1;
    return n * factorial(n - 1);
}

File: Python/recursion/factorial.py
def factorial(n: int) -> int:
    if n in (0, 1):
        return 1
    return n * factorial(n - 1)


def main():
    n = int(input("Enter the Number -> "))
    print(f"Factorial of '{n}' is -> '{factorial(n)}'")


if __name__ == "__main__":
    main()
